### PR TITLE
Fix returning more pageviews with a visit property filter

### PR DESCRIPTION
**Pull request by RobertJoonas**
### Changes

This PR fixes https://github.com/plausible/analytics/issues/1666. The bug was that we were not querying the CollapsingMergeTree table engine correctly, hence the number of pageviews (also custom events) was wrongly inflated when looking at recent periods (`today`, `realtime`).

As the 'sessions' table is using the CollapsingMergeTree engine, we have to select session_id's distinctively in the join subquery. Otherwise we will get multiple rows (with sign -1 and 1) as long as the background merge hasn't happened.

### Tests
- [x] Automated tests have been added

### Changelog
- [x] Not sure whether to log this change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible/stats/base.ex [hunk 1]
```diff
@@ -14,7 +14,7 @@ defmodule Plausible.Stats.Base do
       sessions_q =
         from(
           s in query_sessions(site, query),
-          select: %{session_id: s.session_id}
+          select: %{session_id: fragment("DISTINCT ?", s.session_id)}
```
Comment (aerosol) [inline lib/plausible/stats/base.ex:17]: Is there a reason for not using `where: s.sign == 1` in `query_sessions/2` instead? 🤔 I would expect this to be more efficient.

Comment (aerosol) [reply lib/plausible/stats/base.ex:17]: Unless `where: s.sign == 1` does not guarantee unique results? 🤔

**Review by aerosol — APPROVED** [review]

**Review by aerosol — COMMENTED** [review]
We'll run some perf comparisons before deploying this

File: lib/plausible/stats/base.ex [hunk 2]
```diff
@@ -148,7 +150,7 @@ defmodule Plausible.Stats.Base do
       from(
         s in "sessions",
         where: s.domain == ^site.domain,
-        where: s.start >= ^first_datetime and s.start < ^last_datetime
+        where: s.start >= ^first_datetime and s.start < ^last_datetime,
```
Comment (aerosol) [inline lib/plausible/stats/base.ex:153]: :thinking: unnecessary change?

Comment (RobertJoonas) [reply lib/plausible/stats/base.ex]: Yup, removed in https://github.com/plausible/analytics/pull/2612/commits/e94fe8a9d9d9ac63dd4927f5aff2c21490a18c11
